src/jit/jitstd/list.h

```diff
@@ -68,6 +68,7 @@ class list
         bool operator!=(const const_iterator& it) const;
         const T& operator*() const;
         const T* operator&() const;
+        const T* operator->() const;
```

Comment:
LSRA currently uses the following patter in a few places:

```
ItemIterator iter = someList.begin();

item* i;
for (i = *iter, ++iter; i != nullptr && !i->someCondition(); i = *iter, ++iter)
{
   i->someMethod();
}
```

These uses have been changed to the following pattern:

```
jiststd::list<item>::iterator i = someList.begin()
for (; i != someList.end() && !i->someCondition(); ++i)
{
   i->someMethod(); // This statement uses jitstd::list<item>::operator-> to access the list entry at the current position
}
```